app/src/main/java/org/oppia/app/drawer/NavigationDrawerFragmentPresenter.kt

```diff
@@ -0,0 +1,102 @@
+import androidx.appcompat.app.ActionBarDrawerToggle
+import androidx.appcompat.widget.Toolbar
+import androidx.drawerlayout.widget.DrawerLayout
+import androidx.fragment.app.Fragment
+import com.google.android.material.navigation.NavigationView
+import org.oppia.app.R
+import org.oppia.app.fragment.FragmentScope
+import org.oppia.app.help.HelpActivity
+import org.oppia.app.home.HomeActivity
+import javax.inject.Inject
+
+/** The presenter for [NavigationDrawerFragment]. */
+@FragmentScope
+class NavigationDrawerFragmentPresenter @Inject constructor(
+  private val fragment: Fragment
+) : NavigationView.OnNavigationItemSelectedListener {
+  private lateinit var navView: NavigationView
+  private lateinit var drawerToggle: ActionBarDrawerToggle
+  private lateinit var drawerLayout: DrawerLayout
+  private var previousMenuItemId: Int? = null
+  fun handleCreateView(inflater: LayoutInflater, container: ViewGroup?): View? {
+    val view: View? = inflater.inflate(R.layout.fragment_drawer, container, false)
+    navView = view!!.findViewById(R.id.nav_view)
+    navView.setNavigationItemSelectedListener(this)
+    fragment.setHasOptionsMenu(true)
+    return view
+  }
+
+  private fun openActivityByMenuItemId(menuItemId: Int) {
+    if (previousMenuItemId != menuItemId && menuItemId != 0) {
+      var intent = Intent(fragment.activity, HomeActivity::class.java)
+      when (menuItemId) {
+        R.id.nav_home -> {
+          intent = Intent(fragment.activity, HomeActivity::class.java)
+        }
+        R.id.nav_help -> {
+          intent = Intent(fragment.activity, HelpActivity::class.java)
+        }
+      }
+      fragment.activity!!.startActivity(intent)
+      fragment.activity!!.finish()
+    } else {
+      drawerLayout.closeDrawers()
+    }
+  }
+
+  /** This function contains the DrawerListener and also set the drawer toggle. */
+  fun setUpDrawer(drawerLayout: DrawerLayout, toolbar: Toolbar, menuItemId: Int) {
+    when (menuItemId) {
+      R.id.nav_home -> {
+        navView.menu.getItem(0).isChecked = true
+      }
+      R.id.nav_help -> {
+        navView.menu.getItem(1).isChecked = true
+      }
+    }
+    this.drawerLayout = drawerLayout
+    previousMenuItemId = menuItemId
+    drawerToggle = object : ActionBarDrawerToggle(
+      fragment.activity,
+      drawerLayout,
+      toolbar,
+      R.string.drawer_open,
+      R.string.drawer_close
+    ) {
+      override fun onDrawerOpened(drawerView: View) {
+        super.onDrawerOpened(drawerView)
+        fragment.activity!!.invalidateOptionsMenu()
+      }
+
+      override fun onDrawerClosed(drawerView: View) {
+        super.onDrawerClosed(drawerView)
+        fragment.activity!!.invalidateOptionsMenu()
+      }
+
+      override fun onDrawerSlide(drawerView: View, slideOffset: Float) {
+        super.onDrawerSlide(drawerView, slideOffset)
+        /** Whenever drawer is visible reduse the opacity of toolbar. */
+        toolbar.alpha = 1 - slideOffset / 2
```

Comment:
Should we fully set the toolbar as visible/invisible in the onDrawOpened & onDrawClosed methods to be sure that it's set correctly?

Reply:
It is tested and getting the below difference with and without changing toolbar.alpha
![2019-12-13 (1)](https://user-images.githubusercontent.com/54615666/70793266-ed46fc00-1dc0-11ea-82f0-95879f6ec5d7.png)
![2019-12-13](https://user-images.githubusercontent.com/54615666/70793271-efa95600-1dc0-11ea-835e-65d769cdcf48.png)
Even though the alpha value is getting updated we are removing this code as there is no significant change...